src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+
+        pretty_list = [template % tuple(data[i:i+per_line])
+                       for i in range(0, len(data) - leftover, per_line)]
+        if leftover:
+            return pretty_list + [last_template % tuple(data[-1*leftover:])]
+        return  pretty_list
+
+    def _title(self):
+        """Return filename without extension to be used as title."""
+        title = "Written by cclib-1.5."
+        if self.jobfilename is not None:
+            return os.path.basename(os.path.splitext(self.jobfilename)[0]) +\
+                    '. ' + title
+        return title
+
+    def _keywords(self):
+        """Return one of GTO, GIAO, CSGT keyword."""
+        # AIMALL currently supports on GTO keyword.
+        return self.keyword
+
+    def _no_of_nuclei(self):
+        """Number of nuclei in the molecule."""
+        return len(self.ccdata.atomnos)
+
+    def _no_of_prims(self):
+        """Number of Primitives."""
+        nprims = 0
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                nprims += self.num_orb[prims[0]] * len(prims[1])
+        return nprims
+
+    def _no_of_mos(self):
+        """Number of occupied MOs."""
+        return int(max(self.ccdata.homos)) + 1
+
+    def _no_of_perturbations(self):
+        """Number of Perturbation.
+
+        This is usually zero.  For GIAO it should be 3
+        (corresponding to Lx, Ly and Lz), and
+        for CSGT it should be 6
+        (corresponding to Lx, Ly, Lz, Px, Py and Pz).
+        """
+        if 'GIAO' in self._keywords():
+            return 3
+        elif 'CSGT' in self._keywords():
+            return 6
+        return 0
+
+    def _nuclear_names(self):
+        """Names of nuclei present in the molecule.
+
+        O1
+        H2
+        H3
+        """
+        return [self.pt.element[Z]+str(i) for i, Z in
+                enumerate(self.ccdata.atomnos, start=1)]
+
+    def _atomic_nos(self):
+        """Atomic numbers of elements."""
+        return [str(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_charges(self):
+        """Nuclear charges."""
+        return ['{:22.11E}'.format(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_coords(self):
+        """Nuclear coordinates in Bohr."""
+        coord_template = '%22.11E %.12E %.12E'
+        to_bohr = lambda x: utils.convertor(x, 'Angstrom', 'bohr')
+        nuc_coords = [coord_template % tuple(to_bohr(coord))
+                      for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%22.11E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge if hasattr(self.ccdata, 'charge') else 0.0
+        if len(homos) > 1:
+            return int(sum(homos)) + 2
+        else:
+            return int((homos[0] + 1)*2 - (charge % 2))
```

Comment:
According to the official .wfx specification given on the AIMALL [website](http://aim.tkgristmill.com/wfxformat.html)

> When an ECP is used, the written nuclear charge is less than the real nuclear charge by the number of core electrons modeled by the ECP.  Thus, for Cl and a valency-only ECP, the atomic number is 17 but the nuclear charge is 7.

I would need to consider `ccdata.coreelectrons` here.